### ISSUE TITLE
Letsencrypt Container Disappears

### ISSUE THREAD

**Issue by mr-handerson** [TechPerplexed/Gooby#172]
So I've been keeping an eye on this, and it happened again. My letsencrypt container has disappeared.

It seems like it tried to update (assuming via watchtower?) but failed and now it no longer exists as a container at all.

I have the following unused image showing in Portainer:

Unused | jrcs/letsencrypt-nginx-proxy-companion:latest | 25 MB | 2021-05-16 17:09:12 (created)

So a couple of questions:

1. How do I get it to recreate without having to do a full rclean?
2. What is causing it to fail/delete and how to prevent it from happening again in the future?

EDIT: For 1. I'm assuming Docker-Compose up nginx-proxy would work? I don't want to try yet in case there is testing to be done to try and determine why it failed in the first place

**Comment by mr-handerson** [comment]
Been doing some research. It looks like the associated project has been updated from nginx proxy companion to acme companion

https://github.com/nginx-proxy/acme-companion

So perhaps an update to the 03-proxy.yaml is due?

**Comment by mr-handerson** [comment]
Finally had some time to test this out myself. Updated to the new acme-companion container as best I could. Everything seems to be up and running as far as I can tell. I couldn't get the healthcheck working on the new container, so I had to disable it.

Now to wait for the next autoupdate of the nginx-proxy to see if it disappears again.

**Comment by TechPerplexed** [comment]
Apologies for the very late reply, life's been a little hectic. I HAVE been following your adventures though and wish I had something more useful to say then "let us know" hahaha!

**Comment by mr-handerson** [comment]
Not a problem, I can relate to hectic, hence the delay in me providing an update too. :)

So no ideas why the second container would fail during an autoupdate? The nginx-proxy container seems to update ok, but the companion container fails and ultimately is removed.

Trying to restore it using docker-compose up didn't seem to work properly either. The process never completed. I had to resort to rclean. Admittedly I didn't investigate if pull/build etc was necessary.

**Comment by mr-handerson** [comment]
Ack, same issue. nginx-proxy container updated earlier today and now the acme-companion container is gone.

**Comment by mr-handerson** [comment]
Still no luck. There was an nginx update yesterday and the companion container disappeared again.

There is an error in the Watchtower logs that I think has identified the issue

```
time="2021-06-07T01:01:01Z" level=info msg="Creating /acme"
time="2021-06-07T01:01:01Z" level=error msg="Error response from daemon: No such container: 0d29c4da09b78f556dab9463314d81d2afa5330426eb75733c13ae61c55759ea"
```

From what I can tell, it is trying to start the companion container before the nginx-proxy container. So that's where my focus is next.

**Comment by mr-handerson** [comment]
So now it seems that I have it set so nginx-proxy starts before the companion container, but I still get the same error as above.

Not sure what to try next at this point, but will keep researching for ideas.

**Comment by mr-handerson** [comment]
I tried explicitly setting the container tag to latest, but that didn't make any difference.

I've now set some additional environment variables for Watchtower in addition to turning on debug logging, to at least try and shed some light on why the automated container update and restart is failing. Now just waiting for the next repository update.

**Comment by mr-handerson** [comment]
It seems like it is looking for the old nginx-proxy container, not the new one. But I don't know why, or how, or where.

I've tried splitting the creation of the nginx-proxy and the companion container into separate yaml, not that that should make a lick of difference, but we'll find out in the next update.

**Comment by TechPerplexed** [comment]
Oof, sorry you're having such a hard time with it! Rclean is already a cron set to run at boot, so it might be easy to just add a monthly schedule to it. Unfortunately (or, depending on your viewpoint, fortunately) I don't seem to run into any of the issues you're having!

**Comment by mr-handerson** [comment]
Ah good to know, thanks. Yes, I'll probably just add a schedule in that case.

I would consider it fortunate. I know it should work more smoothly than it is for me, I was just hoping to be able to determine why it isn't and correct it, and that has eluded me thus far.

Given all the issues I had last time I did a full wipe and restore, I'm hesitant to dive in to that again. Plus because I've now modified a lot of the yaml to upgrade to the new acme companion container, and also updated to the current Portainer version, I'm not sure how that would work for a doing a fresh install. I think it would work, since the restore should put my yaml back in? But I need to sit down and actually think through all the steps before I proceed.

So for now I think I will persevere, and do some more research to see if I can determine just what is going on. I did think of a couple more things I can attempt. Hoping to find the time to try those before the next update is released so I can test.

**Comment by mr-handerson** [comment]
In an interesting turn of events, the last update... worked. I will definitely be waiting for another update or two before I get too excited, but fingers crossed.

Conveniently, my certs are also due for renewal soon, so I'll be able to find out if things are actually functioning properly in that regards.

**Comment by mr-handerson** [comment]
OK, so I have now survived a second update, and a certificate renewal. Still don't know why it stopped working in the first place, but glad I was able to come up with a working solution.

Hopefully no one else runs into the issue, but if they do just ping me here and I can post the details.

I will close this for now, and hope that I'm finally back to a stable system that doesn't need any manual intervention.
